gateway/src/main/java/io/zeebe/gateway/impl/job/ActivateJobsHandler.java

```diff
@@ -97,14 +105,19 @@ private void activateJobs(
           response -> false);
     } else {
       // enough jobs activated or no more partitions left to check
-      jobTypeToNextPartitionId.put(jobType, partitionIdIterator.getCurrentPartitionId());
-      responseObserver.onCompleted();
+      try {
+        responseObserver.onCompleted();
+      } catch (Exception e) {
+        // Cannot close the stream. Nothing to do
+      }
     }
   }
 
   private PartitionIdIterator partitionIdIteratorForType(
       final String jobType, final int partitionsCount) {
-    final Integer nextPartitionId = jobTypeToNextPartitionId.computeIfAbsent(jobType, t -> 0);
```

Comment:
Random realization - shouldn't `jobTypeToNextPartitionId` be thread-safe? Right now it's just a normal hash map? Or am I missing something? Isn't this handler potentially called from multiple threads?